nautobot/extras/views.py

```diff
@@ -458,81 +457,27 @@ class CustomFieldBulkDeleteView(generic.BulkDeleteView):
-        if request.POST.get("_all"):
-            if self.filterset is not None:
-                pk_list = [obj.pk for obj in self.filterset(request.GET, model.objects.only("pk")).qs]
-            else:
-                pk_list = model.objects.values_list("pk", flat=True)
-        else:
-            pk_list = request.POST.getlist("pk")
-
-        form_cls = self.get_form()
-
-        if "_confirm" in request.POST:
-            form = form_cls(request.POST)
-            if form.is_valid():
-                logger.debug("Form validation was successful")
-
-                # Delete objects
-                queryset = self.queryset.filter(pk__in=pk_list)
-                try:
-                    tasks = self.construct_custom_field_delete_tasks(queryset=queryset)
-                    chain(*tasks).apply_async()
-                    deleted_count = queryset.delete()[1][model._meta.label]
-                except ProtectedError as e:
-                    logger.info("Caught ProtectedError while attempting to delete objects")
-                    handle_protectederror(queryset, request, e)
-                    return redirect(self.get_return_url(request))
-
-                msg = "Deleted {} {}".format(deleted_count, model._meta.verbose_name_plural)
-                logger.info(msg)
-                messages.success(request, msg)
-                return redirect(self.get_return_url(request))
+    def perform_pre_delete(self, request, queryset):
+        class TaskFailure(Exception):
+            """
+            Custom Exception for when a celery task / tasks failed to execute.
+            """
 
-            else:
-                logger.debug("Form validation failed")
+            pass
 
-        else:
-            form = form_cls(
-                initial={
-                    "pk": pk_list,
-                    "return_url": self.get_return_url(request),
-                }
-            )
-
-        # Retrieve objects being deleted
-        table = self.table(self.queryset.filter(pk__in=pk_list), orderable=False)
-        if not table.rows:
-            messages.warning(
-                request,
-                "No {} were selected for deletion.".format(model._meta.verbose_name_plural),
-            )
-            return redirect(self.get_return_url(request))
-
-        context = {
-            "form": form,
-            "obj_type_plural": model._meta.verbose_name_plural,
-            "table": table,
-            "return_url": self.get_return_url(request),
-        }
-        context.update(self.extra_context())
-        return render(request, self.template_name, context)
+        for obj in queryset:
+            try:
+                result = delete_custom_field_data.delay(obj.name, set(obj.content_types.values_list("pk", flat=True)))
+                # Wait until the custom_field_data is completely deleted
+                while result.ready() is False:
+                    continue
```

Comment:
This is going to loop pretty fast which could cause some I/O contention. It might be better wait a brief interval instead of continuing. Alternatively [`result.wait()`](https://docs.celeryq.dev/en/stable/reference/celery.result.html?highlight=ready(#celery.result.AsyncResult.wait)) may be better as it automatically unblocks when the result is ready, and defaults to `interval=0.5` and supports a `timeout`.